packages/react/src/index.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { createContext, useContext } from 'react'
+import {
+  AnalyticsBrowser,
+  AnalyticsBrowserSettings,
+  InitOptions,
+} from '@segment/analytics-next'
+
+export const createClient = (
+  settings: AnalyticsBrowserSettings,
```

Comment:
Why the decision a createClient function wrapper versus just exporting AnalyticsBrowser (I’ve seen this re-export done in some other react SDKs such as Anplitude’s, and it seemed logical).

Reply:
I was going for maximum compliance with our React Native library as seen here - https://segment.com/docs/connections/sources/catalog/libraries/mobile/react-native/#usage

> ...my inclination would have that (and options as well) as props on “Provider”, instead of making a user always handle client creation themself

"Externalizing" the segment client and then passing it in has benefits. First is that it allows the "Root" component to access it (the one that is wrapping `AnalyticsProvider`), second it allows for setting the client via a factory function, like I'm doing in the Inspector. Similar to `createStore` from Redux.

Reply:
> First is that it allows the "Root" component to access it (the one that is wrapping AnalyticsProvider),

In this rare case they could create their own intermediate component or instantiate the client as an escape valve — per the examples.

> second it allows for setting the client via a factory function, like I'm doing in the Inspector. Similar to createStore from Redux.

Not sure I see the benefit — I don’t  think comparing redux is a good point of comparison — in redux, you aren’t passing the store down — you’re passing “dispatch” — the store is essentially treated like a configuration object. I would compare this to other react wrappers and similar SDKs.

Reply:
The API compliance with our React Native library and factory functions are still a stronghold I believe.

```ts
const getSegmentClient = () => {
  const client = createClient({ ... })
  // setup middlewares
  // setup plugins
  // pre-identify the user

  return client
}
```

cc @chrisradek @pooyaj - You're welcome to shed your opinions on this subject

Reply:
My 2 cents: I think there's value in being consistent with our patterns with regards to the react-native library and in being able to pass a client directly to the provider.

As far as whether it should be required - my intuition is that it makes sense to support passing in just the `writeKey` instead of a client - anecdotally this is what I ended up doing in my own provider. That said, I think this PR is meant to get the code checked in for the published RC library. I think it's worth reaching out to the mobile team to see if they've gotten any feedback on supporting just writeKey and create an issue asking for feedback.

Reply:
Not sure I see the benefit of copying the react native API, especially if it makes things more complicated? Users aren't going to be running our code in a react-native environment, and  there are significant differences:
- React native doesn't use our code at all, `getSegmentClient` is not a wrapper around AnalyticsBrowser.load -- it's a total replacement. If we wanted to ape their API, we'd refactor `analytics-next` to use that.
- React native analytics doesn't use peer dependencies at all.
- React native will never be run on the server, so instantiating in the global scope is always going to make sense.

Reply:
> as far as whether it should be required - my intuition is that it makes sense to support passing in just the writeKey instead of a client -
> That said, I think this PR is meant to get the code checked in for the published RC library.

If so, we will want to hold off on the docs change and the changeset.

Reply:
> Not sure I see the benefit of copying the react native API, especially if it makes things more complicated? Users aren't going to be running our code in a react-native environment, and there are significant differences

It doesn't make things more complicated. It is less code for us to maintain, and better for the end-users as they have an opportunity to augment the client if need be. When there are less disparities between web and mobile libraries, it is better for our customers as their on-boarding experience will be much smoother and quicker, when they are going from web to react-native or vice versa. It'll be second nature.

> React native doesn't use our code at all, `getSegmentClient` is not a wrapper around `AnalyticsBrowser.load` -- it's a total replacement. If we wanted to ape their API, we'd refactor analytics-next to use that.

`getSegmentClient` was a demo of how `createClient` can be augmented further before exposing it to the app through `AnalyticsProvider`. The public API of React Native lib is `createClient` and `AnalyticsProvider`, and so is now in case of React library.

> React native analytics doesn't use peer dependencies at all

We can very well make analytics-react be like this as well. It's just a matter of choice. Plus installation procedure compliance is not the goal here, API compliance is.

> React native will never be run on the server, so instantiating in the global scope is always going to make sense.

Please elaborate

> Ultimately, I think we should take inspiration from other libraries that are more similar to ours

But that'd mean straying away from the consistency of our own "sibling" libraries. I'm not opposing your suggestions, I'm just striving for maximum parity. I would be happy to involve someone who manages react-native library, and see their outlook on making `AnalyticsProvider` "direct" by just passing in `writeKey`.

Reply:
Not strongly opinionated but my two cents on consistency between libs: don't mind that much about consistency over correctness here. They are two different TS based libraries with different architecture, use-cases, etc. So trying to make the interface consistent doesn't add value if it makes using `analytics-react` harder/confusing. We shouldn't be bound to how React Native lib was built originally. 

AFAIK, the reason the React Native library supports passing in a client is that the library supports running multiple instances of the analytics. So you can have multiple providers to track across multiple write-keys. We don't support that at the moment in AJS, so I don't see someone creating multiple clients and use multiple providers. I'm definitely +1 on having the option to pass in the client ( for reasons Neek explained + maybe someone wants to use React library but load AJS through snippet 🤔 ) but whether to make it the only way te use the react library is something to think about.